tests/test_yaml_file_types.py

```diff
@@ -0,0 +1,930 @@
+from contextlib import contextmanager
+
+import pytest
+from ruamel.yaml.constructor import DuplicateKeyError
+
+from komodo.yaml_file_types import (
+    KomodoException,
+    PackageStatusFile,
+    ReleaseFile,
+    RepositoryFile,
+    _komodo_error,
+)
+
+VALID_PYTHON_PACKAGE_KV_LIST = 'zopfli: "0.3"\npytest: 0.40.1'
+INVALID_PYTHON_PACKAGE_KV_LIST_DUPLICATE_KEYS = 'zopfli: "0.3"\nzopfli: "0.3"'
+INVALID_PYTHON_PACKAGE_KV_LIST_FLOAT_PACKAGE_NAME = '1.2: "0.3"\nzopfli: "0.3"'
+INVALID_PYTHON_PACKAGE_KV_LIST_FLOAT_PACKAGE_VERSION = 'zopfli: 0.3\npytest: "0.3"'
+INVALID_PYTHON_PACKAGE_KV_LIST_NONE_PACKAGE_VERSION = 'zopfli:\npytest: "0.3"'
+INVALID_PYTHON_PACKAGE_KV_LIST_UPPERCASE_PACKAGE_NAME = 'zopfli: "0.3"\nPYTEST: "0.3"'
+INVALID_PYTHON_PACKAGE_KV_LIST_MULTIPLE_ERRORS = 'zopfli: 0.3\nPYTEST: "0.3"'
+INVALID_YAML = "zopfli"
+
+
+@contextmanager
+def does_not_raise():
+    yield
+
+
+@pytest.mark.parametrize(
+    "content, expectations",
+    [
+        pytest.param(
+            VALID_PYTHON_PACKAGE_KV_LIST, does_not_raise(), id="valid_release_file"
+        ),
+        pytest.param(
+            INVALID_YAML,
+            pytest.raises(
+                AssertionError,
+                match=r"The file you provided does not appear to be a release file",
+            ),
+            id="invalid_release_file_format",
+        ),
+        pytest.param(
+            INVALID_PYTHON_PACKAGE_KV_LIST_DUPLICATE_KEYS,
+            pytest.raises(DuplicateKeyError, match='found duplicate key "zopfli"'),
```

Comment:
Why do we have a regular exception only here, but all others raise a `SystemExit`?

Reply:
DuplicateKeyError is a ruamel.yaml exception raised when the yaml parser fails due to duplicate keys. That is the error which was thrown at a later stage in the release process, and which I want linted away early on.

Reply:
But should we try to capture the DuplicateKeyError and reraise it as a SystemExit for consistency?

Reply:
Yes, we can absolutely do that